src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Sql/Sprocs/GetInstancesByStudyAndWatermark.sql

```diff
@@ -0,0 +1,43 @@
+/**************************************************************/
+--
+-- STORED PROCEDURE
+--     GetInstancesByStudyAndWatermark
+--
+-- DESCRIPTION
+--     Get instances by given minimum watermark in a study.
+--
+-- PARAMETERS
+--     @batchSize
+--         * The desired number of instances per batch. Actual number may be smaller.
+--     @partitionKey
+--         * The system identified of the data partition.
+--     @studyInstanceUid
+--         * The study instance UID.
+--     @maxWatermark
+--         * The optional maxWatermark.
+-- RETURN VALUE
+--     The instance identifiers.
+------------------------------------------------------------------------
+CREATE OR ALTER PROCEDURE dbo.GetInstancesByStudyAndWatermark
```

Comment:
Maybe it will help me understand if we have the Job workflow. But are you planning to get batches inside a studyUID?

Other options I am thinking is, 
1. Get all the instances for a list of StudyUIDs and save it in the Job and then batch process them in multiple parallel activities.

This will help you optmize for batch size?

Reply:
That will be too many instances, the pattern we follow is split into multiple activities and each activity can be retried. Also where do you suggest to store in the list. Is it in the table? I think we have some limitations on the size of each entry.